### PR TITLE
Windows: prevent NZB names with forbidden elements

### PR DESCRIPTION
**Pull request by shypike**
Replaces #770

### CODE REVIEW

**Comment by thezoggy** [comment]
example reported via email a few weeks back on 1.1.1, can see we sanitize the folder but not the actual nzb name:
`Saving \\?\C:\Users\test\downloads\incomplete\_Con.Man.S02E01.What.Goes.Up.7\__ADMIN__\Con.Man.S02E01.What.Goes.Up.720p.WEB-DL.DD5.1.H.264-CM.nzb.gz failed `

**Comment by Safihre** [comment]
@shypike Maybe we should move the filesystem functions out of `misc.py` to a seperate `filesystem.py`? `misc.py` is getting big..

**Comment by shypike** [comment]
@Safihre Yes, but not in 1.2.1

**Comment by Safihre** [comment]
# For future reference:
Windows does not support paths >256 and has special reserved words.
Programs handle this differently:

- Python uses `\\.\` and this works fine
- `Par2tbb`/`par2cmdline`: Paths should start with the alternative long-path notation: `\\.\`
- `Unrar`: Does not understand `\\?\` or `\\.\` for path to RAR, but does understand and it for the extraction path.
- `Par2` and `unrar` can/will create 'bad' names, so after each step we need to clean the output

**Comment by shypike** [comment]
@Safihre Python uses \\?\ and this works fine.

**Comment by thezoggy** [comment]
windows: reserve 
https://msdn.microsoft.com/en-us/library/windows/desktop/aa365247%28v=vs.85%29.aspx?f=255&MSPPError=-2147217396

https://msdn.microsoft.com/en-us/library/windows/desktop/aa365247%28v=vs.85%29.aspx?f=255&MSPPError=-2147217396#namespaces

File: sabnzbd/newsunpack.py [hunk 1]
```diff
@@ -1192,6 +1171,7 @@ def PAR_Verify(parfile, parfile_nzf, nzo, setname, joinables, classic=False, sin
     stup, need_shell, command, creationflags = build_command(command)
 
     # par2 wants to see \\.\ paths on Windows
+    # See: https://github.com/sabnzbd/sabnzbd/pull/771
```
Comment (shypike) [inline sabnzbd/newsunpack.py]: Dit soort links werken misschien later niet meer

Comment (thezoggy) [inline sabnzbd/newsunpack.py]: `This kind of links might later work no more` 
yes, we should just make a hidden wiki page for dev notes? and then just link to the dev note  ID there instead?

Comment (shypike) [inline sabnzbd/newsunpack.py]: yes

Comment (shypike) [inline sabnzbd/newsunpack.py]: Or just more comments or commit text

**Comment by Safihre** [comment]
Turns out we were partly wrong, par2cmdline doesn't actually support the `\\.\` or `\\?\` notation. It allows to specify the par2-file in this way, but it ignores any of the other files in the folder:
````
Ignoring non-existent source file: \\.
```
So this means that it will fail on obfuscated posts. 
We have to implement a work-a-round so that it doesn't freeze in case of a special Windows word.

**Pull request merged**